lib/ts/recipe/recipeRoutes/index.tsx

```diff
@@ -0,0 +1,130 @@
+import React from "react";
+import NormalisedURLPath from "supertokens-web-js/lib/build/normalisedURLPath";
+import { RoutingComponent } from "../../components/routingComponent";
+import { getSuperTokensRoutesForReactRouterDom } from "../../components/superTokensRoute";
+import { getSuperTokensRoutesForReactRouterDomV6 } from "../../components/superTokensRouteV6";
+import { BaseFeatureComponentMap, ComponentWithRecipeAndMatchingMethod } from "../../types";
+import { getCurrentNormalisedUrlPath } from "../../utils";
+import { RecipeFeatureComponentMap } from "../../types";
+
+export abstract class RecipeRoutes {
+    private pathsToFeatureComponentWithRecipeIdMap?: BaseFeatureComponentMap;
+
+    private static reactRouterDom?: { router: { Route: any }; useHistoryCustom: () => any };
+    private static reactRouterDomIsV6: boolean | undefined = undefined;
+
+    static getRoutes(reactRouterDom: any, instance?: RecipeRoutes): JSX.Element[] {
+        if (instance === undefined) {
```

Comment:
Since we are throwing if the instance param is `undefined`, I'd prefer if it was a required param and we didn't need this check.